### PR TITLE
[Bombastic Perks] Two gun-related perks

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Little revolver-related perk that i thought about
#### Describe the solution
Add perk, add required infra
Also added second perk, that benefits .22 guns
#### Testing
image
image
test of weakpoint enchantment:
no weakpoint enchantment
image
with wp enchantment (numbers are inflated for test purposes)
image
no AP and AP enchantment applied
image
#### Additional context
~~Also thought about perk that gives bonuses for using .22, as a tribute to Holli~~

### CODE REVIEW

File: data/mods/BombasticPerks/perks.json [hunk 1]
```diff
@@ -1178,5 +1178,25 @@
...
+    "id": "revolvers_are_cool",
+    "name": { "str": "Double Action Action" },
+    "description": "You always knew nothing can beat the good old six shooters.  You have a minor bonus using revolvers.",
+    "points": 0,
+    "purifiable": false,
+    "valid": false,
+    "category": [ "perk" ],
+    "enchantments": [
+      {
+        "condition": { "and": [ { "u_has_wielded_with_flag": "RELOAD_ONE" }, { "u_has_wielded_with_skill": "pistol" } ] },
```
Comment (Holli-Git) [inline data/mods/BombasticPerks/perks.json:1193]: Might want to add another flag like SIMPLE_CLEAN or NEVER_JAMS because this would include the mauser

Comment (GuardianDll) [reply data/mods/BombasticPerks/perks.json:1193]: That's how code assign classes tho
I guess it means the game thinks Mauser is revolver
I thought about jsonifying class field, but i don't really know what benefit would it bring

Comment (Holli-Git) [reply data/mods/BombasticPerks/perks.json:1193]: Yeah, seems more like an issue with the mauser tbh. Not a huge issue anyways

**Comment by TheShadowFerret** [comment]
>Also thought about perk that gives bonuses for using .22, as a tribute to Holli

That does what? +AP 🤔?

**Comment by GuardianDll** [comment]
\> That does what? +AP 🤔?

honesly i'd like something more interesting, more engaging, not flat bonuses
but my brain is smol and fails me to make something interesting
armor pen could work

**Comment by Holli-Git** [comment]
> honesly i'd like something more interesting, more engaging, not flat bonuses but my brain is smol and fails me to make something interesting armor pen could work

Make it "bounce around the targets internals" (stun/extra bleed)
Give it the best stopping power (knockdown on hit)
Tactical tininess (make it much more likely to hit weakpoints)
Low power let down (All shots do 20% less damage, with a 20% chance to deal double/triple damage)

**Comment by GuardianDll** [comment]
when i meant "something more engaging" i mean it in literal sense: "make a headshot to deal 2x damage" or stuff like this, something that gonna force player to make *decisions*: do i want to continue and increase the score, or i'll better retreat?

Your ideas are cool tho, i may implement them

**Comment by Holli-Git** [comment]
I apologize for this unsolicited name suggestion but "Rimfire Rebel" has been stuck in my brain

File: data/mods/BombasticPerks/perks.json [hunk 2]
```diff
@@ -1178,5 +1178,41 @@
...
+          { "value": "RANGE", "add": 4 }
+        ]
+      }
+    ]
+  },
+  {
+    "type": "mutation",
+    "id": "22_is_cool",
+    "name": { "str": "Twenty two weakpoints" },
+    "description": "You always appreciated the size of this little fella named \"Dot Two Two Long Rifle\", but they laughed at you.  Not anymore!  You found that small size of a bullet is perfect to hit into little spots between armor or narrow places.  Gives additional armor penetration and quadruple your chances to hit a weakpoint if you use .22 weapon.",
```
Comment (TheShadowFerret) [inline data/mods/BombasticPerks/perks.json:1206]:
```suggestion
    "name": { "str": "Rimfire Rebel" },
    "description": "You always appreciated the size of this little fella named \"Dot Two Two Long Rifle\", but they laughed at you.  Not anymore!  You found that the size of a bullet that small is perfect to hit into little spots between armor or narrow places.   .22 weapons have additional armor penetration and quadrupled chance to hit a weakpoint.",
```

**Comment by TheShadowFerret** [comment]
I don't think altering vanilla weakpoints for mod content is considered kosher 🤔

**Comment by GuardianDll** [comment]
I didn't alter them tho, i added new field, it may be used later for different purposes

**Comment by GuardianDll** [comment]
just add some revolver in vanilla, that fit our inclusion criteria, that can be considered OLD_GUN in the mod, and here you have it

File: data/mods/BombasticPerks/perks.json [hunk 3]
```diff
@@ -1178,5 +1178,41 @@
...
+          { "value": "RANGE", "add": 4 }
+        ]
+      }
+    ]
+  },
+  {
+    "type": "mutation",
+    "id": "22_is_cool",
+    "name": { "str": "Rimfire Rebel" },
+    "description": "You always appreciated the size of this little fella named \"Dot Two Two Long Rifle\", but they laughed at you.  Not anymore!  You found that the size of a bullet that small is perfect to hit into little spots between armor or narrow places.   .22 weapons have additional armor penetration and quadrupled chance to hit a weakpoint.",
```
Comment (GuardianDll) [inline data/mods/BombasticPerks/perks.json:1206]:
```suggestion
    "description": "You always appreciated the size of this little fella named \"Dot Two Two Long Rifle\", but they laughed at you.  Not anymore!  You found that the size of a bullet that small is perfect to hit into little spots between armor or narrow places.  .22 weapons have additional armor penetration and quadrupled chance to hit a weakpoint.",
```

**Pull request merged**